src/libsync/discoveryphase.cpp

```diff
@@ -159,14 +159,20 @@ QPair<bool, QByteArray> DiscoveryPhase::findAndCancelDeletedJob(const QString &o
     QByteArray oldEtag;
     auto it = _deletedItem.find(originalPath);
     if (it != _deletedItem.end()) {
-        ENFORCE((*it)->_instruction == CSYNC_INSTRUCTION_REMOVE
-            // re-creation of virtual files count as a delete
-             || ((*it)->_type == ItemTypeVirtualFile && (*it)->_instruction == CSYNC_INSTRUCTION_NEW)
-             || ((*it)->_isRestoration && (*it)->_instruction == CSYNC_INSTRUCTION_NEW)
-               );
-        (*it)->_instruction = CSYNC_INSTRUCTION_NONE;
-        result = true;
-        oldEtag = (*it)->_etag;
+        const csync_instructions_e instruction = (*it)->_instruction;
+        if (instruction == CSYNC_INSTRUCTION_IGNORE && (*it)->_type == ItemTypeVirtualFile) {
+            result = true;
```

Comment:
I think the result needs to be false here.  We must not consider this as a delete anymore since it is not.
On the other hand, the calling code will assume that it is a move regardless so this may not matter actually.

Either way, i think this deserve a comment.

Reply:
I added an explaining comment.
Returning false triggers actual action, the placeholder file gets removed. See https://github.com/owncloud/client/pull/7814/files#diff-04e3bc4ea34b9a9a993967604f3d74faR924